pkg/crd/crdgeneration/tf2crdgeneration.go

```diff
@@ -67,12 +68,17 @@ func GenerateTF2CRD(sm *corekccv1alpha1.ServiceMapping, resourceConfig *corekccv
 		}
 	}
 
-	statusJSONSchema, err := k8s.RenameStatusFieldsWithReservedNamesIfResourceNotExcluded(resource, statusJSONSchema)
-	if err != nil {
-		return nil, fmt.Errorf("error renaming status fields with reserved names for %#v: %w", statusJSONSchema, err)
+	var err error
+	if k8s.HasComputedFieldsUnderObservedState(resourceConfig.Kind, sm.GetVersionFor(resourceConfig)) {
+		moveComputedFieldsToObservedState(statusOrObservedStateJSONSchema)
+	} else {
+		statusOrObservedStateJSONSchema, err = k8s.RenameStatusFieldsWithReservedNamesIfResourceNotExcluded(resource, statusOrObservedStateJSONSchema)
+		if err != nil {
+			return nil, fmt.Errorf("error renaming status fields with reserved names for %#v: %w", statusOrObservedStateJSONSchema, err)
+		}
 	}
-	populateObservedState(resourceConfig, specJSONSchema, statusJSONSchema)
-	for k, v := range statusJSONSchema.Properties {
+	addObservedFieldsToObservedState(resourceConfig, specJSONSchema, statusOrObservedStateJSONSchema)
```

Comment:
Does this mean we should NOT add a "computed field" to  `observedFields` in servicemapping.

Reply:
Yes. This is indicated in the [observedState field comment](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/pkg/apis/core/v1alpha1/servicemapping_types.go#L185) and executed in the [CRD generation logic](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/pkg/crd/crdgeneration/tf2crdgeneration.go#L464) but I think we should add a unit test to protect it.

Reply:
#1401